blog/models.py

```diff
@@ -68,6 +82,61 @@ def get_absolute_url(self) -> str:
         :rtype: str
         """
         return reverse('ministrant-detail', kwargs={'pk': self.pk})
+    
+    @property
+    def variable_symbol(self):
+        return f'{self.get_actual_camp_year()}{self.pk:04d}'
+
+    @property
+    def unicode_name(self) -> str:
+        return unidecode(f'{self.surname}_{self.birthname}')
+
+    def get_actual_camp_year(self) -> int:
+        return SummerCampInfo.objects.first().start_date.year
+
+    def generate_qr_pay_code(self):
+        qr = qrcode.QRCode(
+            version=None,
+            error_correction=qrcode.constants.ERROR_CORRECT_L,
+            box_size=10,
+            border=1,
+        )
+        print(self.collect_qr_data())
+
+        qr.add_data(self.collect_qr_data())
+        qr.make(fit=True)
+
+        img = qr.make_image(fill_color="black", back_color="white")
+
+        buf = BytesIO()
+        img.save(buf, format='PNG')
+
+        # Rewind the file.
+        buf.seek(0)
+        os.makedirs(os.path.join(settings.MEDIA_ROOT, 'qr_codes'), exist_ok=True)
+        self.qr_pay_code.save(f'{self.unicode_name}.png', ContentFile(buf.getvalue()), save=True)
+
+        return img
+    
+    def collect_qr_data(self):
+        prefix_number = BankAccount.objects.first().number
+        surfix_number = BankAccount.objects.first().bank_code
+    
+        account_number = f'{prefix_number}{surfix_number}' 
+        
+        summer_camp_price = SummerCampInfo.objects.first().price
+        qr_msg = unidecode(f'{self.surname}_{self.birthname}')
+
+        return f'SPD*1.0*ACC:{account_number}*AM:{summer_camp_price}*CC:CZK*MSG:{qr_msg}*X-VS:{self.variable_symbol}*X-KS:0308'
+
+    
+    def save(self, *args, **kwargs):
+        super().save(*args, **kwargs)  # Call the "real" save() method.
+        if not self.qr_pay_code:
+            img = self.generate_qr_pay_code()
+            self.qr_pay_code = f'qr_codes/{self.unicode_name}.png'
```

Comment:
I'd suggest to use the [pathlib module](https://docs.python.org/3/library/pathlib.html) for the paths.